crates/net/nat/src/lib.rs

```diff
@@ -213,7 +220,17 @@ async fn resolve_external_ip_url_res(url: &str) -> Result<IpAddr, ()> {
 
 async fn resolve_external_ip_url(url: &str) -> Option<IpAddr> {
     let response = reqwest::get(url).await.ok()?;
-    let response = response.error_for_status().ok()?;
+    let response = match response.error_for_status() {
+        Ok(response) => response,
+        Err(err) => {
+            let status = err.status().unwrap_or(StatusCode::INTERNAL_SERVER_ERROR);
+            error!(
+                "HTTP error when attempting to resolve external IP using {}: status code {}",
+                url, status
+            );
+            return None;
+        }
+    };
```

Comment:
this is not the only error that can happen here, I find this very invasive and questionable benefit

if you want to log, then this should be rewritten so that we can log the error in one place

Reply:
Happy to remove this if we don't think it's valuable 👍

Reply:
Removed